src/sentry/web/forms/__init__.py

```diff
@@ -9,7 +9,18 @@ class NewNoteForm(forms.Form):
         widget=forms.Textarea(attrs={"rows": "1", "placeholder": "Type a note and press enter..."})
     )
 
-    def save(self, group, user, event=None):
+    def save(self, group, user):
+        qs = Activity.objects.filter(
+            group=group,
+            project_id=group.project_id,
+            user_id=user.id,
+            type=ActivityType.NOTE.value,
+            data=self.cleaned_data,
+        )
+        # Prevent duplicate comments, this is necessary for outbox based
```

Comment:
@markstory should we push this logic to the manager?

Reply:
We could move it there, I didn't want to move too much around at once though.